python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -255,21 +255,26 @@ def _load_single_input(
         return obj
 
     def _load_partition_from_path(
-        self, context: InputContext, path: "UPath", backcompat_path: Optional["UPath"] = None
+        self,
+        context: InputContext,
+        partition_key: str,
```

Comment:
the partition key should actually correspond to the specific partition being loaded, not to the run partition